cpp/src/join/nested_loop_join.cuh

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/detail/utilities/cuda.cuh>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/table/table_view.hpp>
+#include <iostream>
+
+#include <join/hash_join.cuh>
```

Comment:
`#include <join/` looks a little unusual to me. I see this in other join-related files as well, but I think either `#include <cudf/join/hash_join.cuh>` or `#include "hash_join.cuh` would be better.

Reply:
I switched to `<cudf/...>`.

Reply:
Oops, that didn't build -- apparently `<cudf/...>` will find files in the `include` directory but not in the `src` directory. I switched to `"hash_join.cuh"` instead.

Reply:
My mistake; I assumed these headers were under `cpp/include/cudf/join`! Headers in the `src` dir makes me a little nervous, but I guess it's ok as long as these are _only_ included by the cpp/cu files in this dir and especially never by any public headers.